.github/workflows/simulator_nightly.yml

```diff
@@ -20,159 +20,42 @@ concurrency:
   group: ${{ github.workflow }}
 
 env:
-  IOTA_REF: "${{ github.event.inputs.iota_ref || 'main' }}"
+  IOTA_REF: "${{ github.event.inputs.iota_ref || 'develop' }}"
   TEST_NUM: "${{ github.event.inputs.test_num || '30' }}"
+  CARGO_TERM_COLOR: always
+  # Disable incremental compilation.
+  #
+  # Incremental compilation is useful as part of an edit-build-test-edit cycle,
+  # as it lets the compiler avoid recompiling code that hasn't changed. However,
+  # on CI, we're not making small edits; we're almost always building the entire
+  # project from scratch. Thus, incremental compilation on CI actually
+  # introduces *additional* overhead to support making future builds
+  # faster...but no future builds will ever occur in any given CI environment.
+  #
+  # See https://matklad.github.io/2021/09/04/fast-rust-builds.html#ci-workflow
+  # for details.
+  CARGO_INCREMENTAL: 0
+  # Allow more retries for network requests in cargo (downloading crates) and
+  # rustup (installing toolchains). This should help to reduce flaky CI failures
+  # from transient network timeouts or other issues.
+  CARGO_NET_RETRY: 10
+  RUSTUP_MAX_RETRIES: 10
+  # Don't emit giant backtraces in the CI logs.
+  RUST_BACKTRACE: short
+  RUST_LOG: off
+  NUM_CPUS: 24
 
 jobs:
   simtest:
```

Comment:
Could it not be a job of `nightly.yml`?

Reply:
Sure, it totally could. Is that what we want to do?

Reply:
I combined them. I also stopped the reusable rust tests workflow from running its simtests as these should be covered by this one